### PR TITLE
Hp gen8 gen9

### PR DESCRIPTION
**Pull request by FLiPp3r90**
I think this is better now. The role is tested on HP Proliant Gen8 and Gen9 servers.

### CODE REVIEW

File: tasks/disk_drive.yml [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 #following expects that your system has only same type disks on thats why this incredible stupid hack
 - name: Identifying firmware RPM for disk drives
-  shell: "yum search `hpssacli ctrl all show config detail | grep 'Model:' |grep -E '(ATA|HP)' | sort -u| awk '{print $NF}'` | grep -i hp-firmware | awk '{print $1}' | grep `uname  -m`"
+  shell: "yum search `{{ ss_utility }} ctrl all show config detail | grep 'Model:' |grep -E '(ATA|HP)' | sort -u| awk '{print $NF}'` | grep -i firmware-hdd | awk '{print $1}' | grep `uname  -m`"
```
Comment (martbhell) [inline tasks/disk_drive.yml:6]: The pattern matching now does: 

```
 grep -i firmware-hdd 
```

instead of 

```
grep -i hp-firmware 
```

Why was this changed?

**Review by martbhell — CHANGES_REQUESTED** [review]
Looks cool!

File: tasks/disk_drive.yml (hunk 1 above)
Comment (FLiPp3r90) [reply tasks/disk_drive.yml:6]: Hi @martbhell,

the firmware packages for disk drives are named firmware-hdd-*.rpm (for RHEL based systems). With the old grep "hp-firmware" no packages can be found. If it's different on other systems we have to implement a further search do identify the naming of the hdd firmware packages to support all systems.

Comment (martbhell) [reply tasks/disk_drive.yml:6]: Hello @FLiPp3r90 ,

which yum repo are you using?

https://downloads.linux.hpe.com/SDR/repo/spp/RHEL/7/x86_64/current/ has lots of RPMs that start with "hp-firmware". 
Anyway, I like the change to use firmware-hdd in the grep because it matches on newer firwmares and seems we have probably not updated our HDD firmwares in a while! So thank you again for this PR!

As an example (I just took a random HDD) the above repo has these:

```
hp-firmware-hdd-fa46c607d6-HPD1-1.1.x86_64.rpm                        2016-08-30 04:32  4.8M  
firmware-hdd-fa46c607d6-HPD2-1.1.x86_64.rpm                           2017-05-02 22:39  5.0M  
firmware-hdd-fa46c607d6-HPD3-1.1.x86_64.rpm                           2018-04-12 13:37  9.4M  
```

We have some new hardware coming in so we should be able to test this without breaking anything important in the next week or two.

**Review by martbhell — APPROVED** [review]

File: tasks/disk_drive.yml (hunk 1 above)
Comment (FLiPp3r90) [reply tasks/disk_drive.yml:6]: I've referred to the HP repo https://downloads.linux.hpe.com/SDR/repo/spp-gen9/2018.11.0/packages/. 
In the RHEL packages are always named like hp-firmware-hdd.*

**Review by martbhell — APPROVED** [review]
Tested this PR on an XL230 Gen9 and it upgarded firmware where possible. Thanks again @FLiPp3r90 !

There's still some improvements available for this role ( #19 ) but this is a good addition.

 - ilo4 : cannot upgrade 2.61 to 2.61 
 - ilo4.i386: cannot write older (2.60 to 2.61)
 - system ROM: u13.i386 (01/22/2018 to 05/21/2018)
 - smart array HBA H240: cannot upgrade 6.60 to 6.60
 - firmware-hdd : cannot upgrade HPG4 to HPG4
 - hp-firmware-hdd: tried to install HPG1 which didn't work
 - firmware-nic-intel: 1G ROM   1.475.0 to 1.1904.0 --. 1.16.0B ?? 
 - firmware-hca-mellanox-vpi-eth-ib: For ConnectX 3: maybe it did something? Unclear
